sources/packages/api/src/route-controllers/institution-locations/institution-location.institutions.controller.ts

```diff
@@ -246,6 +259,86 @@ export class InstitutionLocationInstitutionsController extends BaseController {
       applicationOfferingName: offering.name,
       applicationProgramDescription: offering.educationProgram.description,
       applicationProgramName: offering.educationProgram.name,
+      applicationProgramCredential: credentialTypeToDisplay(
+        offering.educationProgram.credentialType,
+      ),
+      applicationProgramDelivery: deliveryMethod(
+        offering.educationProgram.deliveredOnline,
+        offering.educationProgram.deliveredOnSite,
+      ),
+      applicationOfferingStudyDelivery: offering.offeringDelivered,
+      applicationOfferingStudyBreak: offering.studyBreaks.map((studyBreak) => ({
+        breakStartDate: dateString(studyBreak.breakStartDate),
+        breakEndDate: dateString(studyBreak.breakEndDate),
+      })),
+      applicationOfferingTuition: offering.actualTuitionCosts,
+      applicationOfferingProgramRelatedCosts: offering.programRelatedCosts,
+      applicationOfferingMandatoryFess: offering.mandatoryFees,
+      applicationOfferingExceptionalExpenses: offering.exceptionalExpenses,
     };
   }
+
+  /**
+   * Save scholastic standing and create new assessment.
+   * @param _locationId location id to check whether the
+   * request user has the permission to this location.
+   * @param applicationId application id.
+   * @param payload Scholastic Standing payload.
+   */
+  @ApiBadRequestResponse({ description: "Invalid form data" })
+  @HasLocationAccess("locationId")
+  @Post(":locationId/application/:applicationId/scholastic-standing")
+  async saveScholasticStanding(
+    @Param("locationId") _locationId: number,
+    @Param("applicationId") applicationId: number,
+    @Body() payload: ScholasticStandingAPIInDTO,
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<void> {
+    try {
+      const submissionResult = await this.formService.dryRunSubmission(
+        FormNames.ReportScholasticStandingChange,
+        payload,
+      );
+
+      if (!submissionResult.valid) {
+        throw new BadRequestException("Invalid submission.");
+      }
+
+      const scholasticStanding =
+        await this.studentScholasticStandingsService.saveScholasticStandingCreateReassessment(
+          applicationId,
+          userToken.userId,
+          submissionResult.data.data,
+        );
+
+      // start assessment.
+      if (scholasticStanding.studentAssessment) {
+        await this.studentAssessmentService.startAssessment(
+          scholasticStanding.studentAssessment.id,
+        );
+      }
+    } catch (error) {
+      switch (error.name) {
+        case APPLICATION_NOT_FOUND:
+        case NOT_A_COMPLETED_APPLICATION:
+          throw new UnprocessableEntityException(
+            new ApiProcessError(error.message, error.name),
+          );
+        default:
+          if (
+            error.message.includes(
```

Comment:
Please use the already in place method to do the validation.
```typescript
await this.studentAssessmentService.assertAllAssessmentsCompleted(
      appealToUpdate.application.id,
);
```
This method already includes a common error message and error name (ASSESSMENT_ALREADY_IN_PROGRESS).

Reply:
Okay. Thanks for that